app/celery/v3/notification_tasks.py

```diff
@@ -105,15 +105,28 @@ def v3_process_notification(request_data: dict, service_id: str, api_key_id: str
         v3_send_email_notification.delay(notification, template)
     elif notification.notification_type == SMS_TYPE:
         if notification.sms_sender_id is None:
-            # Get the template or service default sms_sender_id.
-            # TODO
-            v3_persist_failed_notification(
-                notification, NOTIFICATION_TECHNICAL_FAILURE, "Default logic for sms_sender_id is not yet implemented."
+            # If id is not present, try to fetch the default one associated with the service
+            query = select(ServiceSmsSender).where(
+                (ServiceSmsSender.service_id == service_id) &
+                ServiceSmsSender.is_default
             )
+            with get_reader_session() as db:
+                try:
+                    sms_sender = db.execute(query).one().ServiceSmsSender
+                    v3_send_sms_notification.delay(notification, sms_sender.sms_sender)
```

Comment:
This shouldn't happen here because it changes the flow.  This function is already called below.

Reply:
done